src/main/java/org/mule/service/http/impl/service/util/RedirectUtils.java

```diff
@@ -51,35 +59,55 @@ public static boolean isRedirected(int statusCode) {
   }
 
   /**
-   * @param requestMethod
    * @param responseStatusCode
-   * @return the original request method if the status code is 301, 307 or 308, or GET if it is 302 or 303.
+   * @return false if the status code is 301, 307 or 308, or true if it is 302 or 303.
    */
-  public static String getMethodForStatusCode(String requestMethod, int responseStatusCode) {
+  private static boolean mustSendAsGet(int responseStatusCode) {
     switch (responseStatusCode) {
       case 301:
       case 307:
       case 308:
-        return requestMethod;
+        return false;
       case 302:
```

Comment:
Shouldn't we consider the strict 302 handling?
https://github.com/mulesoft/grizzly-ahc/blob/support/1_14-mule/src/main/java/com/ning/http/client/providers/grizzly/AhcEventFilter.java#L793

Reply:
Yes, i have added that with a test